modifiers.c

```diff
@@ -0,0 +1,111 @@
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include <drm_fourcc.h>
+
+#include "tables.h"
+
+static const char *amd_tile_version_str(uint64_t tile_version) {
+	switch (tile_version) {
+	case AMD_FMT_MOD_TILE_VER_GFX9:
+		return "GFX9";
+	case AMD_FMT_MOD_TILE_VER_GFX10:
+		return "GFX10";
+	case AMD_FMT_MOD_TILE_VER_GFX10_RBPLUS:
+		return "GFX10_RBPLUS";
+	}
+	return "Unknown";
+}
+
+static const char *amd_tile_str(uint64_t tile, uint64_t tile_version) {
+	switch (tile_version) {
+	case AMD_FMT_MOD_TILE_VER_GFX10_RBPLUS:
+		/* fallthrough */
+	case AMD_FMT_MOD_TILE_VER_GFX10:
+		/* fallthrough */
+	case AMD_FMT_MOD_TILE_VER_GFX9:
+		switch (tile) {
+		case AMD_FMT_MOD_TILE_GFX9_64K_S:
+			return "GFX9_64K_S";
+		case AMD_FMT_MOD_TILE_GFX9_64K_D:
+			return "GFX9_64K_D";
+		case AMD_FMT_MOD_TILE_GFX9_64K_S_X:
+			return "GFX9_64K_S_X";
+		case AMD_FMT_MOD_TILE_GFX9_64K_D_X:
+			return "GFX9_64K_D_X";
+		case AMD_FMT_MOD_TILE_GFX9_64K_R_X:
+			return "GFX9_64K_R_X";
+		}
+	}
+	return "Unknown";
+}
+
+static bool amd_gfx9_tile_is_x_t(uint64_t tile) {
+	switch (tile) {
+	case AMD_FMT_MOD_TILE_GFX9_64K_S_X:
+	case AMD_FMT_MOD_TILE_GFX9_64K_D_X:
+	case AMD_FMT_MOD_TILE_GFX9_64K_R_X:
+		return true;
+	}
+	return false;
+}
+
+static void print_amd_modifier(uint64_t mod) {
+	printf("AMD(");
+
+	uint64_t tile_version = AMD_FMT_MOD_GET(TILE_VERSION, mod);
+	uint64_t tile = AMD_FMT_MOD_GET(TILE, mod);
+	uint64_t dcc = AMD_FMT_MOD_GET(DCC, mod);
+	uint64_t dcc_retile = AMD_FMT_MOD_GET(DCC_RETILE, mod);
+
+	printf("TILE_VERSION = %s", amd_tile_version_str(tile_version));
+	printf(", TILE = %s", amd_tile_str(tile, tile_version));
+
+	if (dcc) {
+		printf(", DCC");
+		if (!dcc_retile && AMD_FMT_MOD_GET(DCC_PIPE_ALIGN, mod)) {
+			printf(", DCC_PIPE_ALIGN");
+		}
+		if (AMD_FMT_MOD_GET(DCC_INDEPENDENT_64B, mod)) {
+			printf(", DCC_INDEPENDENT_64B");
+		}
+		if (AMD_FMT_MOD_GET(DCC_INDEPENDENT_128B, mod)) {
+			printf(", DCC_INDEPENDENT_128B");
+		}
+		if (AMD_FMT_MOD_GET(DCC_MAX_COMPRESSED_BLOCK, mod)) {
+			printf(", DCC_MAX_COMPRESSED_BLOCK");
```

Comment:
This is not a boolean (0 = 64 bytes, 1 = 128 bytes, 2 = 256 bytes).

Of course I just noticed that the mask in the headers is 0x1 and the next bit contains another field ...

Reply:
Updated to use the enum, even if drm-next still has the broken version of this.